### PR TITLE
Kernel: Reworked the new kernel synchronization primitives.

### PR DESCRIPTION
**Pull request by Subv**
Properly implemented svcArbitrateLock, svcArbitrateUnlock, svcWaitProcessWideKey and svcSignalProcessWideKey.

These implementations were based on reverse engineering. We should write a suite of hwtests to completely verify every edge case.

I didn't see any regressions in my games, but please test it as much as possible.

### CODE REVIEW

File: src/core/hle/kernel/thread.h [hunk 1]
```diff
@@ -109,6 +109,19 @@ class Thread final : public WaitObject {
      */
     void BoostPriority(u32 priority);
 
+    /*
```
Comment (bunnei) [inline src/core/hle/kernel/thread.h]: nit: These aren't docstrings

**Comment by bunnei** [comment]
Tested a bunch of games, didn't find any obvious regresssions
